flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterConfiguration.java

```diff
@@ -102,21 +102,21 @@ public int getNumTaskManagers() {
 	public String getJobManagerBindAddress() {
 		return commonBindAddress != null ?
 				commonBindAddress :
-				configuration.getString(JobManagerOptions.ADDRESS, "localhost");
+				configuration.getString(JobManagerOptions.BIND_HOST, "localhost");
 	}
 
 	public String getTaskManagerBindAddress() {
 		return commonBindAddress != null ?
 				commonBindAddress :
-				configuration.getString(TaskManagerOptions.HOST, "localhost");
+				configuration.getString(TaskManagerOptions.BIND_HOST, "localhost");
 	}
 
-	public String getJobManagerBindPortRange() {
-		return String.valueOf(configuration.getInteger(JobManagerOptions.PORT, 0));
+	public int getJobManagerBindPort() {
+		return configuration.getInteger(JobManagerOptions.RPC_BIND_PORT, 0);
 	}
 
-	public String getTaskManagerBindPortRange() {
-		return configuration.getString(TaskManagerOptions.RPC_PORT);
+	public int getTaskManagerBindPort() {
+		return configuration.getInteger(TaskManagerOptions.RPC_BIND_PORT, 0);
```

Comment:
Not sure if I fully understand your concern.

I think the RPC services are meant to be used by Flink components internally. Then how would user be affected if the configured address is not used internally?

I have no problem preserving the external address/ports if configured. Just trying to understand.

Reply:
I think the problem I see is that we are using `RPC_BIND_PORT` and `BIND_HOST` as the external address and port when creating the `RpcService` in the `MiniCluster`. It might not be a big problem but it is a behaviour inconsistency.